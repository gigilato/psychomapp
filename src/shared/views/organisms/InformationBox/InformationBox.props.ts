import { IBoxProps } from '$atoms'

export interface IInformationBoxProps extends IBoxProps {
  label: string
  value?: string
}
