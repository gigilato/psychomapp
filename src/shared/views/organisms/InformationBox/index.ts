export * from './InformationBox'
export * from './InformationBox.props'
