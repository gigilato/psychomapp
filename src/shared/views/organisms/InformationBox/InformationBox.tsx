import { Skeleton } from 'moti/skeleton'

import { Box, Text } from '$atoms'

import { IInformationBoxProps } from './InformationBox.props'

export const InformationBox = ({ label, value, children, ...props }: IInformationBoxProps) => {
  return (
    <Box {...props}>
      <Text variant="boldBody" className="mb-xxs">
        {label}
      </Text>
      {children ?? (
        <Skeleton colorMode="dark" width="60%">
          <Text variant="body">{value}</Text>
        </Skeleton>
      )}
    </Box>
  )
}
