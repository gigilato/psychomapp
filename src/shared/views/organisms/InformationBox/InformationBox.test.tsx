import { render } from '@testing-library/react-native'
import React from 'react'

import { InformationBox } from './InformationBox'

describe('InformationBox', () => {
  it('renders correctly', () => {
    const component = render(<InformationBox label="label" value="value" />)
    expect(component).toMatchSnapshot()
  })
})
