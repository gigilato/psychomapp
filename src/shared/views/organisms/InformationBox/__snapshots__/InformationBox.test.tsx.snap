// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InformationBox renders correctly 1`] = `
<View
  style={
    [
      [
        undefined,
        undefined,
      ],
    ]
  }
>
  <Text
    style={
      [
        {
          "fontFamily": "SpaceGrotesk-Medium",
        },
        {
          "fontSize": 16,
        },
        {
          "lineHeight": 18,
        },
        {
          "marginBottom": 4,
        },
      ]
    }
  >
    label
  </Text>
  <Text
    style={
      [
        {
          "fontFamily": "SpaceGrotesk-Regular",
        },
        {
          "fontSize": 16,
        },
        {
          "lineHeight": 18,
        },
      ]
    }
  >
    value
  </Text>
</View>
`;
